namer/namer.cc

```diff
@@ -203,11 +203,18 @@ class SymbolFinder {
         foundMethod.flags = method.flags;
         foundMethod.parsedArgs = ast::ArgParsing::parseArgs(method.args);
         foundMethod.arityHash = ast::ArgParsing::hashArgs(ctx, foundMethod.parsedArgs);
-        foundDefs->addMethod(move(foundMethod));
+        auto def = foundDefs->addMethod(move(foundMethod));
 
         // After flatten, method defs have been hoisted and reordered, so instead we look for the
         // keep_def / keep_self_def calls, which will still be ordered correctly relative to
         // visibility modifiers.
+        ownerStack.emplace_back(def);
+        methodVisiStack.emplace_back(nullopt);
+    }
+
+    void postTransformMethodDef(core::Context ctx, ast::ExpressionPtr &tree) {
+        methodVisiStack.pop_back();
+        ownerStack.pop_back();
```

Comment:
The recent changes ("ensure that fields always have a class owner") mean that `FoundField`s have a `FoundClass` owner when they are entered, rather than having whatever is at the top of the owner stack.

We need methods in the owner stack so we can select the correct context for defining fields (e.g. are we in a singleton method context or an instance method context), even if we're going to ensure that fields are owned by classes, not methods.  At the moment, it also seems easier to have methods in the owner stack so we can select the correct context for other things, but I am open to changing that.

I guess there should be tests for attempting to define various kinds of things (type members, static fields, etc.) inside methods, since namer has never had to think about those sorts of things before.